### PR TITLE
Let UriEndpoint ToString return the Uri

### PR DESCRIPTION
**Pull request by ste1nstone**
Summary of the changes (Less than 80 chars)
 - Overriding the ToString Method of UriEndpoint to return the ToString Result if the Uri Property

Addresses #12656

### CODE REVIEW

File: src/Servers/Connections.Abstractions/src/UriEndPoint.cs [hunk 1]
```diff
@@ -24,5 +24,7 @@ public UriEndPoint(Uri uri)
         /// The <see cref="System.Uri"/> defining the <see cref="EndPoint"/>.
         /// </summary>
         public Uri Uri { get; }
+
+        public override string ToString() => Uri.ToString();
```
Comment (Tratcher) [inline src/Servers/Connections.Abstractions/src/UriEndPoint.cs]: @davidfowl do you want the encoded or decoded result? ToString is the decoded display value. AbsoluteUri is the encoded useful value.

Comment (davidfowl) [reply src/Servers/Connections.Abstractions/src/UriEndPoint.cs]: Decoded seems fine.

**Review by halter73 — APPROVED** [review]

**Comment by pranavkm** [comment]
@ste1nstone think you might need to update ref assemblies. Could you rin the instructions provided in https://github.com/aspnet/AspNetCore/blob/master/docs/ReferenceAssemblies.md#when-changing-public-api in the  `src/Servers/Connections.Abstractions/src` directory?

**Comment by halter73** [comment]
Hi @ste1nstone. Sorry for the delay merging your PR. There's still one more reference assembly that needs to be updated before we can merge the PR. You can either rerun `.\eng\scripts\GenerateReferenceAssemblies.ps1` yourself, or you can cherry-pick https://github.com/aspnet/AspNetCore/commit/4baa717811a9c396060fe337e75bd056198d4347. Thanks!

**Comment by ste1nstone** [comment]
Hi @halter73 no problem.
Hope that no everything is OK.

**Comment by halter73** [comment]
Thanks @ste1nstone!

**Pull request merged**